tests/installation/grub_test.pm

```diff
@@ -55,7 +55,12 @@ sub handle_installer_medium_bootup {
             $self->handle_uefi_boot_disk_workaround;
         }
         else {
+            # we need to assure that we boot from the disk
+            unless (check_screen 'inst-bootmenu-boot-harddisk') {
+                send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
```

Comment:
Still I think you should not need that. `send_key_until_needlematch` should check first if we are not already on the first entry so basically then we end up with the same code as in line L53

Reply:
I don't see an alternative to touch the other branches